### ISSUE TITLE
phenomatch.jar not working with newest hpo.obo releases

### ISSUE THREAD

**Issue by czepeda0** [ibn-salem/position_effect#5]
Hi Jonas,

The phenomatch.jar script is not working with the newest releases of hpo.obo. I get the following error:

Oct 11, 2021 6:02:36 PM ontologizer.go.OBOParser doParse
INFO: Got 11785 terms and 15459 relations in 61 ms
Details of parsed obo file:
  date:                 null
  format:               1.2
  term definitions:     11785
Oct 11, 2021 6:02:36 PM ontologizer.go.Ontology assignLevel1TermsAndFixRoot
INFO: Ontology contains a single level-one term (All (HP:0000001)
WARN: Frequency of 4576 terms was zero!! Calculated by -1 * (Math.log(1/(double)maxFreq)) = -1 * (Math.log(1/(double)3525)))
Set IC of these to : 8.16763571524637
[INFO] Ontology and annotation table were parsed.
WARNING term is null:
HP:0410030
Exception in thread "main" java.io.IOException: [ERROR] this term 'HP:0410030'. is null. Probably it was not found in the provied phenotype ontology.
        at phenotypeontology.PhenotypeData.getTermIncludingAlternatives(PhenotypeData.java:535)
        at io.TabFileParser.parseCNVwithHpoTerms(TabFileParser.java:384)
        at de.charite.compbio.phenomatch.core.Phenomatch.<init>(Phenomatch.java:100)
        at de.charite.compbio.phenomatch.commandline.PhenomatchCMD.main(PhenomatchCMD.java:27)

Can this be fixed?
Thanks!
Cinthya

**Comment by czepeda0** [comment]
@ibn-salem See above open issue :)

**Comment by ibn-salem** [comment]
Hi @czepeda0 

thanks a lot for bringing this up! 

I finally found a bit of time to look into it. However, I cannot reproduce the error. 

I downloaded the latest `hp.obo` file from here: http://purl.obolibrary.org/obo/hp.obo
This link is provided here: https://hpo.jax.org/app/download/ontology

According to the header of the file, it is version: `hp/releases/2021-10-10`

```
format-version: 1.2
data-version: hp/releases/2021-10-10
saved-by: Peter Robinson, Sebastian Koehler, Sandra Doelken, Chris Mungall, Melissa Haendel, Nicole Vasilevsky, Monarch Initiative, et al.
subsetdef: hposlim_core "Core clinical terminology"
subsetdef: secondary_consequence "Consequence of a disorder in another organ system."
```
And with this file, I could also run the tool ` java -jar bin/phenomatch.jar` on the provided example data. 

The terms that lead to the error in your case seem to be `HP:0410030`. This term is included in the latest `hp.obo` file.

Can you please check the version and source of your `.obo` file?
And can you check if the term `HP:0410030` is contained in it?

Thank you!

**Comment by czepeda0** [comment]
Thanks Jonas! Upgrading the hp.obo file did solve the issue, however this now shows up in two different computers used to run the code:

```
Nov 03, 2021 11:46:53 AM ontologizer.go.OBOParser doParse
INFO: Got 16601 terms and 20529 relations in 469 ms
Details of parsed obo file:
  date: null
  format: 1.2
  term definitions: 16601
Nov 03, 2021 11:46:53 AM ontologizer.go.Ontology assignLevel1TermsAndFixRoot
INFO: Ontology contains a single level-one term (All (HP:0000001)
Exception in thread "main" java.lang.RuntimeException: Can't handle annotation-file format!
at phenotypeontology.PhenotypeData.readAnnotations(PhenotypeData.java:379)
at phenotypeontology.PhenotypeData.<init>(PhenotypeData.java:133)
at de.charite.compbio.phenomatch.core.Phenomatch.<init>(Phenomatch.java:91)
at de.charite.compbio.phenomatch.commandline.PhenomatchCMD.main(PhenomatchCMD.java:27)
```

Any idea where this might be coming from? I updated my computer modules, dependencies, etc to make sure it's not something wrong with my computer, but I keep getting the same result.

Thanks!
Cinthya

**Comment by ibn-salem** [comment]
this is fixed now in version 0.0.2
